src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,36 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder result = new StringBuilder("Report for period " + dateFrom + " - " + dateTo);
+        int[] salary = new int[names.length];
```

Comment:
we don't need this array, just add salary to string builder after it's counted for each name

Reply:
this array stores salary data. and is used in a loop. I can't do it as you wrote, I don't know how.
![image](https://user-images.githubusercontent.com/98239700/212396409-616e2bd0-7d40-4646-9c67-f2ce3ed700be.png)

Reply:
![image](https://user-images.githubusercontent.com/98239700/212396873-23d33437-3581-41ea-8cf2-2d676cd74560.png)

Reply:
It's because you iterate lines first and then names. If you change the order - for (names) then for (lines) - you will have all information for e.g. John at the end of one iteration of the outer loop, You will write John's salary directly to StringBuilder, hence eliminating need for a third loop and a salary array